src/adapters/supabase/helpers/chats.ts

```diff
@@ -0,0 +1,122 @@
+import { SupabaseClient } from "@supabase/supabase-js";
+import { Super } from "./supabase";
+import { Context } from "../../../types/context";
+
+export type Chat = {
+  chatId: number;
+  chatName: string;
+  taskNodeId: string;
+  status: string;
+};
+
+/**
+ * Handles all telegram chat storage and retrieval
+ */
+export class Chats extends Super {
+  constructor(supabase: SupabaseClient, context: Context) {
+    super(supabase, context);
+  }
+
+  /**
+   * Used for storing the userIds of a chat before we
+   * ban all of them and archive the chat.
```

Comment:
Very curious to experience this and see if the vibe is as expected when kicked. 

I don't recall if it actually deletes the group from our inbox or if we see the last message. Looking forward to testing it.

Reply:
I think it deletes from inbox. Other services, like Instagram, you can see the group chat and see the latest message "X person removed you from the group" and thats what is also visible in your inbox preview, which is not preferred for this workgroup experience.